WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignListView.swift

```diff
@@ -0,0 +1,140 @@
+                viewModel.loadCampaigns()
+            }
+            let webViewController = AuthenticatedWebViewController(viewModel: webViewModel)
+            self?.navigationController?.show(webViewController, sender: self)
+        }
+    }
+
+    @available(*, unavailable)
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+/// View for showing a list of campaigns.
+///
+struct BlazeCampaignListView: View {
+    @ObservedObject private var viewModel: BlazeCampaignListViewModel
+    @State private var selectedCampaignURL: URL?
+
+    var onCreateCampaign: () -> Void = {}
+
+    private let siteURL: String
+
+    init(viewModel: BlazeCampaignListViewModel, siteURL: String) {
+        self.viewModel = viewModel
+        self.siteURL = siteURL
+    }
+
+    var body: some View {
+        Group {
+            switch viewModel.syncState {
+            case .results:
+                RefreshableInfiniteScrollList(spacing: Layout.contentSpacing,
+                                              isLoading: viewModel.shouldShowBottomActivityIndicator,
+                                              loadAction: viewModel.onLoadNextPageAction,
+                                              refreshAction: { completion in
+                    viewModel.onRefreshAction(completion: completion)
+                }) {
+                    ForEach(viewModel.campaigns) { item in
+                        BlazeCampaignItemView(campaign: item)
+                            .onTapGesture {
+                                let path = String(format: Constants.campaignDetailsURLFormat,
+                                                  item.campaignID, siteURL,
+                                                  BlazeCampaignSource.campaignList.rawValue)
+                                selectedCampaignURL = URL(string: path)
+                            }
+                    }
+                }
+            case .empty:
+                EmptyState(title: Localization.emptyStateTitle,
+                           description: Localization.emptyStateMessage,
+                           image: .emptyProductsImage)
+                    .frame(maxHeight: .infinity)
+            case .syncingFirstPage:
+                ActivityIndicator(isAnimating: .constant(true), style: .medium)
+            }
+        }
+        .padding(Layout.contentSpacing)
+        .navigationTitle(Localization.title)
+        .navigationBarTitleDisplayMode(.inline)
+        .toolbar {
+            ToolbarItem(placement: .confirmationAction) {
+                Button(Localization.create) {
+                    onCreateCampaign()
+                }
+            }
+        }
+        .onAppear {
+            viewModel.loadCampaigns()
+        }
+        .sheet(item: $selectedCampaignURL) { url in
+            detailView(url: url)
+        }
+    }
+}
+
+private extension BlazeCampaignListView {
```

Comment:
Super nit: Consider keeping this extension for View-related code and creating a new extension for constants and strings.